cfme/scripting/quickstart.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import print_function
+
+import os
+import sys
+import argparse
+import subprocess
+import hashlib
+
+IS_SCRIPT = sys.argv[0] == __file__
+CWD = os.getcwd()  # we expect to be in the workdir
+REDHAT_BASED = os.path.isfile('/etc/redhat-release')
+CREATED = object()
+REQUIREMENT_FILE = 'requirements/frozen.txt'
+
+REDHAT_INSTALL_COMMAND = (
+    "pkcon install -yq"
+    " python-virtualenv gcc postgresql-devel libxml2-devel"
+    " libxslt-devel zeromq3-devel libcurl-devel"
+    " redhat-rpm-config gcc-c++ openssl-devel"
+    " libffi-devel python-devel")
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--mk_virtualenv", default="../cfme_venv")
+parser.add_argument("--system-site-packages", action="store_true")
+parser.add_argument("--config_path", default="../cfme-qe-yamls/complete/")
+
+
+def command_text(command, shell):
+    if shell:
+        return command
+    else:
+        command = [repr(x) if ' ' in x else x for x in command]
+        return ' '.join(command)
+
+
+def call_or_exit(command, shell=False):
+    try:
+        print('QS $', command_text(command, shell))
+        res = subprocess.call(command, shell=shell)
+    except Exception as e:
+        print(repr(e))
+        sys.exit(1)
+    else:
+        if res:
+            print("call failed with", res)
+            sys.exit(res)
+
+
+def install_system_packages():
+    if REDHAT_BASED:
+        call_or_exit(REDHAT_INSTALL_COMMAND, shell=True)
+    else:
+        print("WARNING: unknown distribution,",
+              "please ensure you have the required packages installed")
+        print("INFO: on redhat based systems this is the equivalend of:")
+        print("$", REDHAT_INSTALL_COMMAND)
+
+
+def setup_virtualenv(target, use_site):
+    if os.path.isdir(target):
+        print("INFO: Virtualenv", target, "already exists, skipping creation")
+        return CREATED
+    add = ['--system-site-packages'] if use_site else []
+
+    call_or_exit(['virtualenv', target] + add)
+
+
+def venv_call(venv_path, command, *args):
+    # pop PYTHONHOME to avoid nested environments
+    os.environ.pop('PYTHONHOME', None)
+    call_or_exit([
+        os.path.join(venv_path, 'bin', command),
+    ] + list(args))
+
+
+def hash_file(path):
+    content_hash = hashlib.sha1()
+    with open(path, 'rb') as fp:
+        content_hash.update(fp.read())
+    return content_hash.hexdigest()
+
+
+def install_requirements(venv_path, quiet=False):
```

Comment:
This install mechanism doesn't seem to take the PYCURL_SSL_LIBRARY env var into account?